src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php

```diff
@@ -120,6 +118,7 @@ public function authenticateUserAndHandleSuccess(UserInterface $user, Request $r
     public function handleAuthenticationFailure(AuthenticationException $authenticationException, Request $request, GuardAuthenticatorInterface $guardAuthenticator, $providerKey)
     {
         $token = $this->tokenStorage->getToken();
+        $this->dispatchEvent(SecurityEvents::INTERACTIVE_LOGIN_FAILURE, new InteractiveLoginFailureEvent($request, $authenticationException));
```

Comment:
Will move this below the next block so it's consistent. In all other cases this is dispatched only after token is erased from tokenStorage.